cmd/spire-server/cli/run/run_posix_test.go

```diff
@@ -4,18 +4,168 @@
+	testTempDir = "/tmp/spire-server-test"
 )
 
+func TestCommand_Run(t *testing.T) {
+	type fields struct {
+		logOptions         []log.Option
+		env                *commoncli.Env
+		allowUnknownConfig bool
+	}
+	type args struct {
+		args []string
+	}
+	type want struct {
+		code           int
+		dataDirCreated string
+		stderrContent  string
+	}
+	tests := []struct {
+		name         string
+		fields       fields
+		args         args
+		configLoaded bool
+		want         want
+	}{
+		{
+			name: "don't create any dir when error loading nonexistent config",
+			args: args{
+				args: []string{},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: new(bytes.Buffer),
+				},
+				allowUnknownConfig: false,
+			},
+			configLoaded: false,
+			want: want{
+				code:          1,
+				stderrContent: "could not find config file",
+			},
+		},
+		{
+			name: "don't create any dir when error loading invalid config",
+			args: args{
+				args: []string{
+					"-config", "../../../../test/fixture/config/agent_run_posix.conf",
+					"-namedPipeName", "\\spire-agent\\public\\api",
+				},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: new(bytes.Buffer),
+				},
+				allowUnknownConfig: false,
+			},
+			configLoaded: false,
+			want: want{
+				code:          1,
+				stderrContent: "flag provided but not defined: -namedPipeName",
+			},
+		},
+		{
+			name: "create data dir when config is loaded and server crashes",
+			args: args{
+				args: []string{
+					"-config", "../../../../test/fixture/config/server_run_posix.conf",
+					"-dataDir", fmt.Sprintf("%s/wrong/data/dir", testTempDir),
+				},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: new(bytes.Buffer),
+				},
+				allowUnknownConfig: false,
+			},
+			configLoaded: true,
+			want: want{
+				code:           1,
+				dataDirCreated: fmt.Sprintf("%s/wrong/data/dir", testTempDir),
+			},
+		},
+		{
+			name: "create data dir when config is loaded and server stops",
+			args: args{
+				args: []string{
+					"-config", "../../../../test/fixture/config/server_run_posix.conf",
+					"-dataDir", fmt.Sprintf("%s/data", testTempDir),
+				},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: new(bytes.Buffer),
+				},
+				allowUnknownConfig: false,
+			},
+			configLoaded: true,
+			want: want{
+				code:           0,
+				dataDirCreated: fmt.Sprintf("%s/data", testTempDir),
+			},
+		},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+			_ = fflag.Unload()
+			os.RemoveAll(testTempDir)
+			defer os.RemoveAll(testTempDir)
+
+			cmd := &Command{
+				logOptions:         testCase.fields.logOptions,
+				env:                testCase.fields.env,
+				allowUnknownConfig: testCase.fields.allowUnknownConfig,
+			}
+
+			go func() {
+				time.Sleep(1 * time.Second)
```

Comment:
Generally speaking, it's not a good idea to sleep the process to wait for an event to occur. Is this to wait that the server loaded completely? Could we inspect the output to see the progress of the loading and have a timeout?

Reply:
Yeah, apparently i can use spiretest lib to inspect the output logs for waiting right?